icu4c/source/test/intltest/numbertest_api.cpp

```diff
@@ -6085,6 +6086,55 @@ void NumberFormatterApiTest::formatUnitsAliases() {
     }
 }
 
+void NumberFormatterApiTest::formatArbitraryConstant() {
+    IcuTestErrorCode status(*this, "formatArbitraryConstant");
+
+    struct TestCase {
+        const char *unitIdentifier;
+        int32_t inputValue;
+        UNumberUnitWidth width;
+        Locale locale;
+        const UnicodeString expectedOutput;
+    } testCases[]{
+        // {"meter-per-kelvin-second", 2, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+        //  "2 meters per second-kelvin"},
+        {"meter-per-100-kelvin-second", 3, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"3 meters per 100-second-kelvin"},
+        {"meter-per-1000", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(), u"1 meter per 1000"},
+        {"meter-per-1000-second", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"1 meter per 1000-second"},
+        {"meter-per-1000-second-kelvin", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"1 meter per 1000-second-kelvin"},
+        {"meter-per-1-second-kelvin-per-kilogram", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"1 meter per 1-kilogram-second-kelvin"},
+        {"meter-second-per-kilogram-kelvin", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"1 meter-second per kilogram-kelvin"},
+        {"meter-second-per-1000-kilogram-kelvin", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"1 meter-second per 1000-kilogram-kelvin"},
+        {"meter-second-per-1000-kilogram-kelvin", 1, UNUM_UNIT_WIDTH_SHORT, Locale::getEnglish(),
+         u"1 m⋅sec/1000⋅kg⋅K"},
+        {"meter-second-per-1000-kilogram-kelvin", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getGerman(),
+         u"1 Meter⋅Sekunde pro 1000⋅Kilogramm⋅Kelvin"},
+        {"meter-second-per-1000-kilogram-kelvin", 1, UNUM_UNIT_WIDTH_SHORT, Locale::getGerman(),
```

Comment:
Mark had been arguing for special formatting for larger denominators, so that if you had "meter-per-1e6-kilogram", you'd get "meter per 1,000,000 kilograms".  Did you do that?  If so, you should have a test for it.

Reply:
unfortunately, for now, it is just a plain number.

There is an issue for that: https://unicode-org.atlassian.net/browse/ICU-23039